src/api/useSearchVenuesOffer/useSearchVenueOffers.ts

```diff
@@ -0,0 +1,130 @@
+    const venueAlreadyListedPrice = offerVenues[venueAlreadyListedIndex]?.price ?? 0
+    if (
+      venueAlreadyListedIndex >= 0 &&
+      hit.offer.prices?.length &&
+      venueAlreadyListedPrice > hit.offer.prices[0]
+    ) {
+      offerVenues[venueAlreadyListedIndex].offerId = Number(hit.objectID)
+      offerVenues[venueAlreadyListedIndex].price = hit.offer.prices[0]
+      return
+    }
+
+    offerVenues.push({
+      offerId: Number(hit.objectID),
+      price: hit.offer.prices?.length ? hit.offer.prices[0] : 0,
+      venueId: hit.venue.id,
+      title: hit.venue.publicName ? hit.venue.publicName : hit.venue.name ?? '',
+      address: formatFullAddress(hit.venue.address, hit.venue.postalCode, hit.venue.city),
+      coordinates: hit._geoloc,
+    })
+  })
+
+  return offerVenues
+}
+
+type FilterVenueOfferType = {
+  hit: AlgoliaHit
+  offerId: number
+  venueId: number | undefined
+}
+
+export const filterVenueOfferHit = ({ hit, offerId, venueId }: FilterVenueOfferType): boolean =>
+  typeof hit.offer.subcategoryId !== 'undefined' &&
+  hit.objectID !== String(offerId) &&
+  hit.venue.id !== venueId
+
+export const useSearchVenueOffersInfiniteQuery = ({
+  offerId,
+  venueId,
+  query,
+  geolocation,
+  shouldExecuteQuery,
+}: UseSearchVenueOffersType) => {
+  const isUserUnderage = useIsUserUnderage()
+  const transformHits = useTransformOfferHits()
+  const { setCurrentQueryID } = useSearchAnalyticsState()
+  const previousPageObjectIds = useRef<string[]>([])
+
+  const { data, ...infiniteQuery } = useInfiniteQuery<Response>(
+    [QueryKeys.SEARCH_RESULTS, { ...initialSearchState, view: undefined, query }],
+    async ({ pageParam: page = 0 }) => {
+      const response = await fetchOffers({
+        parameters: { ...initialSearchState, query, page, hitsPerPage: 10 },
+        userLocation: geolocation,
+        isUserUnderage,
+        storeQueryID: setCurrentQueryID,
+        excludedObjectIds: previousPageObjectIds.current,
+        isFromOffer: true,
+      })
+
+      previousPageObjectIds.current = response.hits.map((hit) => hit.objectID)
+      return response
+    },
+    // first page is 0
+    { getNextPageParam, enabled: shouldExecuteQuery }
+  )
+
+  const offerVenues = useMemo(() => {
+    const flattenedHits = flatten(data?.pages.map((page) => page.hits.map(transformHits))).filter(
+      (hit) => filterVenueOfferHit({ hit, offerId, venueId })
+    ) as Offer[]
+
+    return getOfferVenues(flattenedHits)
+  }, [data?.pages, offerId, transformHits, venueId])
+
+  const nbOfferVenues = offerVenues.length
+
+  return { data, offerVenues, nbOfferVenues, ...infiniteQuery }
+}
+
+export const useSearchVenueOffers = ({
+  offerId,
+  venueId,
+  query,
+  geolocation,
+  shouldExecuteQuery,
+}: UseSearchVenueOffersType) => {
+  return useSearchVenueOffersInfiniteQuery({
+    offerId,
+    venueId,
+    query,
+    geolocation,
+    shouldExecuteQuery,
+  })
+}
```

Comment:
Deux choses :
* Si tu en as vraiment besoin tu peux juste faire ça puisque ce sont exactement les même paramètres (`UseSearchVenueOffersType`)
```suggestion
export const useSearchVenueOffers = useSearchVenueOffersInfiniteQuery
```
* Mais surtout je vois pas l'intérêt du coup... C'est encore plus vrai quand on voit ma suggestion raccourcie. Je te propose donc de simplement ne pas exporter 2 constantes mais une seule.

Reply:
Yes c'est vrai